### PR TITLE
Update Pill to follow the new button styles

### LINKED ISSUE #3786: Multi select combo box pill hover is wrong in theme next
### Package name(s)

Core (@salt-ds/core)

### Package version(s)

1.31.0

### Description

In theme next, pill within combo box is light blue on hover and is not readable

https://storybook.saltdesignsystem.com/?path=/story/core-combo-box--multiselect&globals=density:touch;themeNext:enable

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Operating system

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [X] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

## Work notes

- Combo box spec should be updated to make it clear which colors are override from default pill
- Design to be done in #3643

### LINKED ISSUE #3930: Update Pill to follow the new actionable tokens introduced by #3325
Spec:

New actionable tokens were added for button but "button" like components need to be updated e.g. Pill

```[tasklist]
### Tasks
- [x] Update pill spec
- [x] Update component in Figma
- [ ] #3925
```

### PR DESCRIPTION
**Pull request by Vineet119**
Closes #3786, closes #3930

### CODE REVIEW

File: packages/core/src/pill-input/PillInput.css [hunk 1]
```diff
@@ -282,21 +282,48 @@
   display: contents;
 }
 
+.saltPillInput .saltPill {
```
Comment (joshwooding) [inline packages/core/src/pill-input/PillInput.css:285]: These changes should be done in Pill.css and only the overrides needed for PillInput should be added here

**Review by joshwooding — CHANGES_REQUESTED** [review]
see comment

**Review by joshwooding — DISMISSED** [review]
The tokens used need to be reviewed

**Review by joshwooding — DISMISSED** [review]

File: packages/core/src/pill-input/PillInput.css [hunk 2]
```diff
@@ -283,20 +283,12 @@
 }
 
 .saltPillInput .saltPill:focus-visible {
-  background: var(--salt-selectable-background-hover);
+  background: var(--salt-content-foreground-highlight);
+  color: var(--salt-content-primary-foreground);
+  border-color: var(--salt-content-foreground-highlight);
```
Comment (joshwooding) [inline packages/core/src/pill-input/PillInput.css:288]:
```suggestion
  border-color: transparent;
```

**Review by yunjungyeh — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**